qa-pipelines/tasks/lib/azure-aks.sh

```diff
@@ -68,45 +78,71 @@ azure_wait_for_lbs_in_namespace() {
     done
 }
 
-azure_set_a_record() {
-    az network dns record-set a create \
-        --resource-group ${AZURE_DNS_RESOURCE_GROUP} \
-        --zone-name ${AZURE_DNS_ZONE_NAME} \
-        --name $1 \
-        --ttl 300
-    az network dns record-set a add-record \
-        --resource-group ${AZURE_DNS_RESOURCE_GROUP} \
-        --zone-name ${AZURE_DNS_ZONE_NAME} \
-        --record-set-name $1 \
-        --ipv4-address $2
+azure_set_record() {
+    if [[ ${cap_platform} == "eks" ]]; then
+        az network dns record-set cname create \
+            --resource-group ${AZURE_DNS_RESOURCE_GROUP} \
+            --zone-name ${AZURE_DNS_ZONE_NAME} \
+            --name $1 \
+            --ttl 300
+
+        lb_hostname=$(echo "${2}" | jq -r '.[] | select(.svc == "'${3}'").hostname')
+
+        az network dns record-set cname set-record \
+            --resource-group ${AZURE_DNS_RESOURCE_GROUP} \
+            --zone-name ${AZURE_DNS_ZONE_NAME} \
+            --record-set-name $1 \
+            --cname $lb_hostname
+    else
+        az network dns record-set a create \
+            --resource-group ${AZURE_DNS_RESOURCE_GROUP} \
+            --zone-name ${AZURE_DNS_ZONE_NAME} \
+            --name $1 \
+            --ttl 300
+
+        lb_ip=$(echo "${2}" | jq -r '.[] | select(.svc == "'${3}'").ip')
+
+        az network dns record-set a add-record \
+            --resource-group ${AZURE_DNS_RESOURCE_GROUP} \
+            --zone-name ${AZURE_DNS_ZONE_NAME} \
+            --record-set-name $1 \
+            --ipv4-address $lb_ip
+    fi
 }
 
 azure_set_record_sets_for_namespace() {
     local namespace=$1
     local lb_info lb_ip
-    lb_info=$(kubectl get svc -n $namespace -o json | jq -c '[.items[] | select(.spec.type == "LoadBalancer") | {"svc": .metadata.name, "ip": .status.loadBalancer.ingress[0].ip}]')
+    lb_info=$(
+      kubectl get svc -n $namespace -o json |
+      jq -c '
+        [
+          .items[] | select(.spec.type == "LoadBalancer") | {
+            "svc": .metadata.name,
+            "ip": .status.loadBalancer.ingress[0].ip,
+            "hostname": .status.loadBalancer.ingress[0].hostname
+           }
+        ]'
+    )
     for lb_svc in $(echo "${lb_info}" | jq -r '.[] | .svc'); do
-        lb_ip=$(echo "${lb_info}" | jq -r '.[] | select(.svc == "'${lb_svc}'").ip')
         if [[ ${lb_svc} == "uaa-uaa-public" ]]; then
-            azure_set_a_record uaa.$AZURE_AKS_RESOURCE_GROUP $lb_ip
-            azure_set_a_record *.uaa.$AZURE_AKS_RESOURCE_GROUP $lb_ip
+            azure_set_record uaa.$AZURE_AKS_RESOURCE_GROUP "${lb_info}" "${lb_svc}"
```

Comment:
I don't think `lb_info` needs to be passed to `azure_set_record`. Can't all the information needed be taken from `lb_svc`?

Reply:
nope. lb_info is needed for ip/hostname

Reply:
Right, I see that you're taking just the `.svc` value from the load balancer service structure. This is unintuitive because you're passing an entire array (`lb_info`) and then one piece of information from the element of the array we care (the service information). Instead, we should just pass the one element from the array. `azure_set_record` *only* needs information about the service it's setting a record for, not all the other service information.

Here's what you can do for the above `for` loop instead (relying on the fact that the compact representation won't have any whitespace that isn't in a string value, and `.svc`, `.ip`, and `.hostname` will never have whitespace either):

```
for lb_svc in $(echo "${lb_info}" | jq -c .[]; do
...
  azure_set_record ... "${lb_svc}"
...
done
```

`azure_set_record` can then start with:
```
lb_name=$(jq .svc <<< $2)
lb_hostname=$(jq .hostname <<< $2)
lb_ip=$(jq .ip <<< $2)
```